### PR TITLE
OSRA426 orphan integration tests

### PR DESCRIPTION
**Pull request by ZmagoD**

### CODE REVIEW

**Comment by gmunumel** [comment]
looks good to me :+1:

File: spec/features/hq/orphan_spec.rb [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+  background do
+    i_sign_in_as_admin
+    an_orphan_exists 1, "Orphan 1", "Father 1", "2012-01-01", "Male", "Normal"
+    an_orphan_exists 2, "Orphan 2", "Father 2", "2011-01-01", "Female", "High"
+  end
+  
+  scenario 'There should be a list of orphans on the admin index page' do
+    visit hq_orphans_path
+    and_i_should_see "Orphan 1 Father 1"
```
Comment (razorcd) [inline spec/features/hq/orphan_spec.rb]: What about checking for the name only first and then checking the other details? And having the father in a different line.
```
and_i_should_see "Orphan 1"
and_i_should_see_within "Orphan 1", "Father 1"
and_i_should_see_within "Orphan 1", "2012-01-01"
.....
and_i_should_see "Orphan 2"
....
```

File: spec/features/hq/orphan_spec.rb [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    check 'orphan[goes_to_school]'
+    select 'Inactive', :from => 'orphan[status]'
+    select 'High', :from => 'orphan[priority]'
+    fill_in 'orphan[family_name]', with: 'Asdf'
+    fill_in 'orphan[mother_name]', with: 'Aisha'
+    check 'orphan[mother_alive]'
+    fill_in 'orphan[father_occupation]', with: 'Merchant'
+    fill_in 'orphan[guardian_name]', with: 'Goku'
+    fill_in 'orphan[guardian_id_num]', with: '999'
+    fill_in 'orphan[guardian_name]', with: '999.999.999'
```
Comment (razorcd) [inline spec/features/hq/orphan_spec.rb]: Shouldn't this be a name?

File: spec/features/hq/orphan_spec.rb [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+    fill_in 'orphan[alt_contact_number]', with: '000999'
```
Comment (razorcd) [inline spec/features/hq/orphan_spec.rb]: `orphan[guardian_id_num] = 999` and `orphan[alt_contact_number] = 000999` are to similar
when checking if we see `999` couldn't botch match? I might be wrong.

File: spec/features/hq/orphan_spec.rb [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+    and_i_should_be_on "hq_orphan_path", { orphan_id: 1 }
+    and_i_should_see "Orphan successfuly saved"
+    and_i_should_see "Orphan N"
+    and_i_should_see "01 January 2010"
+    and_i_should_see 'Asdf'
+    and_i_should_see 'Aisha'
+    and_i_should_see 'Merchant'
+    and_i_should_see 'Rome'
+    and_i_should_see '999'
+    and_i_should_see '000999'
```
Comment (razorcd) [inline spec/features/hq/orphan_spec.rb]: we should really check all fields are present in the show view

File: spec/features/hq/orphan_spec.rb (hunk 2 above)
Comment (ZmagoD) [inline spec/features/hq/orphan_spec.rb]: What name of the orphan?

Comment (razorcd) [inline spec/features/hq/orphan_spec.rb]: `guardian_name` is `999.999.999`, shouldn't this be a name?

Comment (ZmagoD) [inline spec/features/hq/orphan_spec.rb]: Oh yes, sorry :)

File: spec/features/hq/orphan_spec.rb [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+    and_i_should_see 'New Town'
+    and_i_should_see 'Homs'
+    and_i_should_see 'NewTownHood'
+    and_i_should_see 'NewTownStreet'
+    and_i_should_see 'NewTown details'
+    and_i_should_see 'Sponsored by Another Org. Yes'
+    and_i_should_see '10'
+    and_i_should_see '5'
+    and_i_should_see 'This is a comment'
+  end
```
Comment (razorcd) [inline spec/features/hq/orphan_spec.rb]: Uncle Bob would not like a method like this :) It is too long and it is doing too many things.
Can we extract some parts in new methods?

Like:
```
visit_orphan( "Orphan 1" )
and_i_click_link "Edit Orphan"
and_i_should_be_on "edit_hq_orphan_path", { orphan_id: 1 }

expect_filled_formfields_for "Orphan 1"  #checks that all formfields have the correct values

update_formfields_for "Orphan 1"  #fills in formfields with new data
and_i_click_button "Update Orphan"

and_i_should_be_on "hq_orphan_path", { orphan_id: 1 }
and_i_should_see "Orphan successfuly saved"
expect_updated_data_for "Orphan 1"  #checks that the new updated data is present
```

**Pull request merged**